src/_components/AssetSpecs/AddAssetSpecContainer.jsx

```diff
@@ -0,0 +1,70 @@
+import React from 'react';
+import { connect } from 'react-redux';
+import PropTypes from 'prop-types';
+import { createAssetSpec } from '../../_actions/assetSpecs.actions';
+import AddAssetSpecComponent from '../../components/AssetSpecs/AddAssetSpecComponent';
+
+class AddAssetSpecContainer extends React.Component {
+  state = {
+    assetSpec: {
+      year_of_manufacture: '',
+      processor_speed: '',
+      screen_size: '',
+      processor_type: '',
+      storage: '',
+      memory: ''
+    }
+  };
+
+  handleInputChange = (event, data = {}) => {
+    const name = event.target.name || data.name;
+    const value = event.target.value || data.value;
+
+    this.setState({
+      assetSpec: {
+        ...this.state.assetSpec,
```

Comment:
If a new state depends on the previous state,  I recommend using an [updater function](https://reactjs.org/docs/react-component.html#setstate) as setting state is an asynchronous action. E.g
```jsx
this.setState(prevState => {
  return {
    ...prevState.assetSpec
   [name]: value
  }
})
```